File: C/exercise/reverseDegree.c

#include "reverseDegree.h"

int reverseDegree(char *s)
{
	int degree = 0;
	for (int i = 0; s[i] != '\0'; i++) {
		degree += (26 - (s[i] - 'a')) * (i + 1);
	}
	return degree;
}

File: Dart/lib/reverseDegree.dart
class Solution {
  int reverseDegree(String s) {
    var product = 0;
    for (var i = 0; i < s.length; i++) {
      product += (26 - s[i].codeUnitAt(0) + 'a'.codeUnitAt(0)) * (i + 1);
    }
    return product;
  }
}
